common/lib/xmodule/xmodule/js/spec/capa/display_spec.coffee

```diff
@@ -333,45 +351,50 @@ describe 'Problem', ->
       @problem = new Problem($('.xblock-student_view'))
 
     it 'log the problem_reset event', ->
-      spyOn($, 'postWithPrefix').andCallFake (url, answers, callback) ->
+      spyOn($, 'postWithPrefix').and.callFake (url, answers, callback) ->
         promise =
           always: (callable) -> callable()
       @problem.answers = 'foo=1&bar=2'
       @problem.reset()
       expect(Logger.log).toHaveBeenCalledWith 'problem_reset', 'foo=1&bar=2'
 
     it 'POST to the problem reset page', ->
-      spyOn($, 'postWithPrefix').andCallFake (url, answers, callback) ->
+      spyOn($, 'postWithPrefix').and.callFake (url, answers, callback) ->
         promise =
           always: (callable) -> callable()
       @problem.reset()
       expect($.postWithPrefix).toHaveBeenCalledWith '/problem/Problem1/problem_reset',
           { id: 'i4x://edX/101/problem/Problem1' }, jasmine.any(Function)
 
     it 'render the returned content', ->
-      spyOn($, 'postWithPrefix').andCallFake (url, answers, callback) ->
-        callback html: "Reset!"
+      spyOn($, 'postWithPrefix').and.callFake (url, answers, callback) ->
+        callback html: "Reset"
         promise =
             always: (callable) -> callable()
       @problem.reset()
-      expect(@problem.el.html()).toEqual 'Reset!'
+      expect(@problem.el.html()).toEqual 'Reset'
 
-    it 'tests if all the buttons are disabled and the text of check button remains same while resetting', ->
-      runs ->
-        spyOn($, 'postWithPrefix').andCallFake (url, answers, callback) ->
-          promise =
-            always: (callable) -> callable()
+    it 'tests if all the buttons are disabled and the text of check button remains same while resetting', (done) ->
+      deferred = $.Deferred()
+      self = this
+
+      runs = ->
+        spyOn($, 'postWithPrefix').and.callFake (url, answers, callback) ->
+          promise = undefined
+          promise = always: (callable) ->
+            callable()
         spyOn @problem, 'enableAllButtons'
         @problem.reset()
         expect(@problem.enableAllButtons).toHaveBeenCalledWith false, false
         expect(@problem.checkButtonLabel).toHaveText 'Check'
-      waitsFor (->
-        return jQuery.active == 0
-      ), "jQuery requests finished", 1000
+        if jQuery.active == 0
+          deferred.resolve()
+        deferred.promise()
 
-      runs ->
-        expect(@problem.enableAllButtons).toHaveBeenCalledWith true, false
-        expect(@problem.checkButtonLabel).toHaveText 'Check'
+      runs.call(self).then(->
+        expect(self.problem.enableAllButtons).toHaveBeenCalledWith true, false
+        expect(self.problem.checkButtonLabel).toHaveText 'Check'
+      ).always done
```

Comment:
I think we can get rid of the alias `self` here and we also don't need to call runs with `self` context it's already there.
we can just use the following syntax

```coffeescript
runs.call(self).then(_.bind((->
  #code
), this)).always done
```